### ISSUE TITLE
pymedphys experimental dicom listen only binds to localhost

### ISSUE THREAD

**Issue by didymo** [pymedphys/pymedphys#1432]
The command:
pymedphys experimental dicom listen -d ~/test_data -a PMPSTORE_SCP 10401

only binds to port 10401 on localhost.

sudo netstat -na | sudo grep 10401
tcp        0      0 127.0.0.1:10401         0.0.0.0:*               LISTEN   

sudo nmap  workstationipnumber
Starting Nmap 7.80 ( https://nmap.org ) at 2021-02-16 08:39 AEDT
Nmap scan report for Lion.lan (workstationipnumber)
Host is up (0.0000020s latency).
Not shown: 996 closed ports
PORT    STATE SERVICE
22/tcp  open  ssh
25/tcp  open  smtp
80/tcp  open  http
443/tcp open  https

Nmap done: 1 IP address (1 host up) scanned in 0.14 seconds

there are no firewalls etc etc to block this.

**Comment by sjswerdloff** [comment]
@pchlap I've only used it on localhost... is this working in your environment across hosts?
Any suggestion?

**Comment by sjswerdloff** [comment]
I see that https://github.com/pymedphys/pymedphys/blob/ae9c2bc7e1da09855bff60c10504dd8e389ab032/lib/pymedphys/_dicom/connect/listen.py#L196
isn't including passing in a host, the CLI doesn't offer an argument for specifying which IP address (on the current host) to bind to, and defaults to binding on localhost (127.0.0.1).

At: https://stackoverflow.com/questions/166506/finding-local-ip-addresses-using-pythons-stdlib#166589
I found the following for finding the one with default routing:
```
import socket
def get_ip():
    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    try:
        # doesn't even have to be reachable
        s.connect(('10.255.255.255', 1))
        IP = s.getsockname()[0]
    except Exception:
        IP = '127.0.0.1'
    finally:
        s.close()
    return IP
```
I don't think I can get to this until the weekend, but I'd be happy to review then or ask for your review if I'm the one who gets to it first.

**Comment by pchlap** [comment]
Thanks @sjswerdloff
I will have a go at solving this today. I'll let you know how I go, if I can get something together for you to review then you can pick it apart over the weekend :)

**Comment by SimonBiggs** [comment]
Fixed in #1434
